Services/AssessmentQuestion/interfaces/interface.ilAsqQuestion.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/* Copyright (c) 1998-2013 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Interface ilAsqQuestion
+ *
+ * @author    Björn Heyser <info@bjoernheyser.de>
+ * @version    $Id$
+ *
+ * @package    Services/AssessmentQuestion
+ */
+interface ilAsqQuestion
+{
+	/**
+	 * @param int $parentId
+	 * @return void
+	 */
+	public function setParentId($parentId);
+	
+	/**
+	 * @return int
+	 */
+	public function getParentId();
+	
+	/**
+	 * @param int $questionId
+	 * @return void
+	 */
+	public function setId($questionId);
+	
+	/**
+	 * @return int $questionId
+	 */
+	public function getId();
+	
+	/**
+	 * @return string
+	 */
+	public function getTitle();
+	
+	/**
+	 * @return string
+	 */
+	public function getComment();
+	
+	/**
+	 * @return string
+	 */
+	public function getOwner();
+	
+	/**
+	 * @return string $questionType
+	 */
+	public function getQuestionType();
+	
+	/**
+	 * @return string
+	 */
+	public function getQuestionText();
+	
+	/**
+	 * @return string
+	 */
+	public function getPoints();
+	
+	/**
+	 * @return string
+	 */
+	public function getEstimatedWorkingTime();
+	
+	/**
+	 * @return void
+	 */
+	public function load();
+	
+	/**
+	 * @return void
+	 */
+	public function save();
+	
+	/**
+	 * @return void
+	 */
+	public function delete();
+	
+	/**
+	 * @param ilQTIItem $qtiItem
+	 * @return void
+	 */
+	public function fromQtiItem(ilQTIItem $qtiItem);
```

Comment:
As far as I understand QTI, this cannot possibly be a member of this interface, because you already need a specific instance implementing the interface, but get the information which instance this might be only via the `ilQTIItem`. I would suggest you consider to move this QTI-serialization/deserialization into a single service as well, similar to persistance. A file containing qti-formatted question even could be considered a case of general persistance for questions.

Reply:
By now, the qti is indeed an own service, and yes, the serialisation to xml should be done there, but I need to only decouple assessment questions from the current consumers by now.

reworking the qti service should be done, i agree, but let us focus on the assessment questions in the first place as i cannot fully reimplement all the things at once.